connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -829,6 +878,34 @@ private boolean startConnector(String connectorName) {
         return started;
     }
 
+    private Callable<Void> getConnectorCallable(final String connectorName, final ConnectorStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startConnector(connectorName);
+                        } catch (ConfigException e) {
```

Comment:
I think this is a throwback from before https://github.com/apache/kafka/pull/1778/files#diff-10752971682d4575c93ddec45e0553f0L755

We shouldn't need to catch any exception here.